test/infer/test_autoguide.py

```diff
@@ -859,3 +860,73 @@ def model2():
     svi = SVI(model2, guide, optim.Adam(0.01), Trace_ELBO())
     with pytest.raises(RuntimeError, match="are no local variables"):
         svi.run(random.PRNGKey(0), 10)
+
+
+def test_autosldais(N=64, D=3, num_steps=45000, num_samples=2000):
+    def _model(X, Y):
+        theta = numpyro.sample(
+            "theta", dist.Normal(jnp.zeros(D), jnp.ones(D)).to_event(1)
+        )
+        with numpyro.plate("N", N, subsample_size=2 * N // 3):
+            X_batch = numpyro.subsample(X, event_dim=1)
+            Y_batch = numpyro.subsample(Y, event_dim=0)
+            numpyro.sample("obs", dist.Bernoulli(logits=theta @ X_batch.T), obs=Y_batch)
+
+    def _surrogate_model(X, Y):
+        theta = numpyro.sample(
+            "theta", dist.Normal(jnp.zeros(D), jnp.ones(D)).to_event(1)
+        )
+        omegas = numpyro.param(
+            "omegas", 2.0 * jnp.ones(N // 2), constraint=dist.constraints.positive
+        )
+
+        with numpyro.plate("N", N // 2), numpyro.handlers.scale(scale=omegas):
+            X_batch = numpyro.subsample(X, event_dim=1)
+            Y_batch = numpyro.subsample(Y, event_dim=0)
```

Comment:
I think we can use `X`, `Y` here for clarity or change the signature of this model to `X_batch`, `Y_batch` and inside the model, replace `N // 2` by `M = Y_batch.shape[0]`.